### PR TITLE
Bugfix - Check version of installed php

### PR DESCRIPTION
**Pull request by rrobgill**
Signed-off-by: Rob Gill <rrobgill@protonmail.com>

**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Not install an old version of PHP if the system already has a newer version installed. 

#2117 identifies this issue.

Condition is somewhat uncommon as it appears to only happen if updated PHP has been installed, not from default repos. New repo offers packages eg ```php7.0*``` , but system contains no (or outdated) metapackage ```php```.

**How does this PR accomplish the above?:**

During installation, checks current version of php. If a presently supported version (5.5 or greater) is found, then it is used as the basis for installing the php related web dependencies (x-common, x-cli, x-sqlite). 

This is achieved by setting existing variable ```phpVer``` based upon currently installed php version. Only code relating to setting ```phpVer``` is changed, there is no change to installation code.

**What documentation changes (if any) are needed to support this PR?:**

None.

---

### CODE REVIEW

**Comment by rrobgill** [comment]
Is stickler applying sh rules here?
Bash definitely supports decimals in this context.

**Comment by dschaper** [comment]
Try using `-lt` instead of the `<` operator.

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -160,13 +160,25 @@ if command -v apt-get &> /dev/null; then
...
+      phpVer="php"
     # fall back on the php5 packages
-    phpVer="php5"
+    else 
+      phpVer="php5"
+    fi
+  else
+    # Supported php is installed, its common, cgi & sqlite counterparts are deps
+    phpInsMajor="$(echo "$phpInsVersion" | cut -d\. -f1)"
+    phpInsMinor="$(echo "$phpInsVersion" | cut -d\. -f2)"
```
[the remark this answers is not in the document]
Comment (rrobgill) [reply automated install/basic-install.sh]: Fixed used on line 181.

File: automated install/basic-install.sh [hunk 2]
```diff
@@ -160,13 +160,25 @@ if command -v apt-get &> /dev/null; then
...
-    phpVer="php"
-  # If not,
-  else
+  # Check for and determine version number, major and minor of current php install
+  if command -v php &> /dev/null; then
+    phpInsVersion="$(php -v | head -n1 | grep -Po '(?<PHP )[^ ]+')"
+    echo -e "  ${INFO} Existing PHP installation detected : PHP version $phpInsVersion"
+  fi
+  # Check if installed php is supported version (5.4 is EOL)
+  if [[ $phpInsVersion < "5.5" ]]; then
```
[the remark this answers is not in the document]
Comment (dschaper) [reply automated install/basic-install.sh]: `if [[ ${phpInsVersion} -lt 5.5 ]]; then`

**Comment by rrobgill** [comment]
Nope, it dies on -gt and -lt. They're for integers only.

**Comment by dschaper** [comment]
Are you trying to do a string comparison and a less than with strings?

**Comment by dschaper** [comment]
Have you tried `(("$a" < "$b"))`?

File: automated install/basic-install.sh [hunk 3]
```diff
@@ -160,13 +160,29 @@ if command -v apt-get &> /dev/null; then
...
-    phpVer="php5"
+    else 
+      phpVer="php5"
+    fi
+  else
+    # Supported php is installed, its common, cgi & sqlite counterparts are deps
+    phpVer="php$phpInsMajor.$phpInsMinor"
   fi
   # We also need the correct version for `php-sqlite` (which differs across distros)
   if ${PKG_MANAGER} install --dry-run ${phpVer}-sqlite3 > /dev/null 2>&1; then
```
Comment (Mcat12) [inline automated install/basic-install.sh]: PHP 5.5 passes the test, but there is no `php5.5-sqlite3` or `php5.5-sqlite` package available in Debian. Only 7.0, 7.1, and 7.2 have packages which include the minor version in the name.

File: automated install/basic-install.sh [hunk 4]
```diff
@@ -160,13 +160,29 @@ if command -v apt-get &> /dev/null; then
...
+    phpInsVersion="$(php -v | head -n1 | grep -Po '(?<=PHP )[^ ]+')"
+    echo -e "  ${INFO} Existing PHP installation detected : PHP version $phpInsVersion"
+    phpInsMajor="$(echo "$phpInsVersion" | cut -d\. -f1)"
+    phpInsMinor="$(echo "$phpInsVersion" | cut -d\. -f2)"
+    # Is installed php version newer than php5?
+    if [ "$(echo "$phpInsMajor.$phpInsMinor < 7.0" | bc )" == 0 ]; then
+      phpInsNewer=true
+    fi
+  fi
+  # Check if installed php is unsupported version (5.4 is EOL)
```
Comment (Mcat12) [inline automated install/basic-install.sh]: This comment needs to be updated

File: automated install/basic-install.sh [hunk 5]
```diff
@@ -160,13 +160,29 @@ if command -v apt-get &> /dev/null; then
...
-    phpVer="php"
-  # If not,
-  else
+  # Check for and determine version number (major and minor) of current php install
+  if command -v php &> /dev/null; then
+    phpInsVersion="$(php -v | head -n1 | grep -Po '(?<=PHP )[^ ]+')"
+    echo -e "  ${INFO} Existing PHP installation detected : PHP version $phpInsVersion"
+    phpInsMajor="$(echo "$phpInsVersion" | cut -d\. -f1)"
+    phpInsMinor="$(echo "$phpInsVersion" | cut -d\. -f2)"
+    # Is installed php version newer than php5?
```
Comment (Mcat12) [inline automated install/basic-install.sh]: This comment should say we're checking if PHP is at least 7.0, which makes the following if statement make more sense.

**Pull request merged**